detoks/src/main/java/nl/tudelft/trustchain/detoks/Strategy.kt

```diff
@@ -40,8 +49,39 @@ private val strategyComparators = mutableMapOf<Int, (Pair<TorrentHandler, Profil
     private fun lowestWatchTimeStrategy(
         p0: Pair<TorrentHandler, ProfileEntry?>,
         p1: Pair<TorrentHandler, ProfileEntry?>
-    ) : Int = p1.second!!.watchTime compareTo p0.second!!.watchTime
+    ) : Int = p0.second!!.watchTime compareTo p1.second!!.watchTime
+
+    /**
+     * Returns the torrent handlers based on number of likes
+     */
+    private fun topFirstStrategy(
+        p0: Pair<TorrentHandler, ProfileEntry?>,
+        p1: Pair<TorrentHandler, ProfileEntry?>
+    ) : Int = p0.second!!.likes compareTo p1.second!!.likes
+
+    /**
+     * Returns the torrent handlers based on when they were first uploaded
+     */
+    private fun newestFirstStrategy(
+        p0: Pair<TorrentHandler, ProfileEntry?>,
+        p1: Pair<TorrentHandler, ProfileEntry?>
+    ) : Int = p0.second!!.uploadDate compareTo p1.second!!.uploadDate
 
+    /**
+     * Returns the torrent handlers based on hopcount
+     */
+    private fun hotStrategy(
+        p0: Pair<TorrentHandler, ProfileEntry?>,
+        p1: Pair<TorrentHandler, ProfileEntry?>
+    ) : Int = p0.second!!.hopCount compareTo p1.second!!.hopCount
+
+    /**
+     * Returns the torrent handlers based on likes / hopcount ratio
+     */
+    private fun risingStrategy(
+        p0: Pair<TorrentHandler, ProfileEntry?>,
+        p1: Pair<TorrentHandler, ProfileEntry?>
+    ) : Int = (p0.second!!.likes / (p0.second!!.hopCount + 1)) compareTo (p1.second!!.likes / (p1.second!!.hopCount + 1))
```

Comment:
I don't think this is the right ratio, maybe freshness cut-off of a couple hour + likes?